test/Transition-test.js

```diff
@@ -334,16 +356,19 @@ describe('Transition', () => {
 
         onEntered() {
           expect(nodeRef.current.textContent).toEqual(`status: ${ENTERED}`);
-          expect(count).toEqual(2);
-          done();
```

Comment:
Should we have to verify this `onEntered` has been called?

Reply:
This is similar to https://github.com/reactjs/react-transition-group/pull/771#discussion_r791570073:

This condition moved down to 

```js
     await waitFor(() => {
        expect(count).toEqual(2);
      });
```